quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/SimpleUserAppServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+
+        APP_1 = new CitesphereAppInfo();
+        APP_1.setClientId(CLIENT_ID_1);
+
+        APP_2 = new CitesphereAppInfo();
+        APP_2.setClientId(CLIENT_ID_2);
+
+        APP_3 = new CitesphereAppInfo();
+        APP_3.setClientId(CLIENT_ID_3);
+    }
+
+    @Test
+    public void test_findByUsername_success() {
+        Mockito.when(simpleUserAppRepository.findByUsername(USER_1)).thenReturn(USER_APP_LIST);
+        List<SimpleUserApp> response = simpleUserAppService.findByUsername(USER_1);
+        Assert.assertNotNull(response);
+        Assert.assertNotEquals(0, response.size());
+        for (SimpleUserApp app : response) {
+            Assert.assertTrue(USER_APP_LIST.stream().anyMatch(userapp -> userapp.getId().equals(app.getId())));
+        }
+    }
+
+    @Test
+    public void test_findByUsername_empty() {
+        Mockito.when(simpleUserAppRepository.findByUsername(USER_2)).thenReturn(new ArrayList<>());
+        List<SimpleUserApp> actualResponse = simpleUserAppService.findByUsername(USER_2);
+        Assert.assertNotNull(actualResponse);
+        Assert.assertEquals(0, actualResponse.size());
+    }
+    
+    @Test
+    public void test_delete_success() {
+        Mockito.when(simpleUserAppRepository.findByUsernameAndAppClientId(USER_1, CLIENT_ID_1)).thenReturn(USER_APP_1);
+        simpleUserAppService.delete(USER_1, CLIENT_ID_1);
+    }
+    
+    @Test
+    public void test_delete_noEntry() {
+        Mockito.when(simpleUserAppRepository.findByUsernameAndAppClientId(USER_1, CLIENT_ID_1)).thenReturn(null);
+        simpleUserAppService.delete(USER_1, CLIENT_ID_1);
+    }
+
+    @Test
+    public void test_getCitesphereApps_allMatching() {
+        List<CitesphereAppInfo> citesphereApps = new ArrayList<>();
+        citesphereApps.add(APP_1);
+        citesphereApps.add(APP_2);
+
+        Mockito.when(simpleUserAppRepository.findByUsername(USER_1)).thenReturn(USER_APP_LIST);
+        Mockito.when(citesphereConnector.getCitesphereApps()).thenReturn(citesphereApps);
+
+        List<CitesphereAppInfo> response = simpleUserAppService.getCitesphereApps(SIMPLE_USER_1);
+
+        for (CitesphereAppInfo app : getOverlappingApps(citesphereApps, USER_APP_LIST)) {
+            Assert.assertTrue(response.stream().anyMatch(responseApp -> responseApp.getClientId().equals(app.getClientId())));
+        }
+    }
+
+    @Test
+    public void test_getCitesphereApps_partialMatching() {
+        List<CitesphereAppInfo> citesphereApps = new ArrayList<>();
+        citesphereApps.add(APP_2);
+        citesphereApps.add(APP_3);
+
+        Mockito.when(simpleUserAppRepository.findByUsername(USER_1)).thenReturn(USER_APP_LIST);
+        Mockito.when(citesphereConnector.getCitesphereApps()).thenReturn(citesphereApps);
+
+        List<CitesphereAppInfo> response = simpleUserAppService.getCitesphereApps(SIMPLE_USER_1);
+        
+        for (CitesphereAppInfo app : getOverlappingApps(citesphereApps, USER_APP_LIST)) {
```

Comment:
this seems overly complicated. Why not just check the length of `response` (=1) and that the object in there is app_2?